src/use_promise_guard/index.es

```diff
@@ -1,19 +1,18 @@
+import { useCallback } from 'react';
 import useMounted from '../use_mounted';
 
 const usePromiseGuard = () => {
   const isMounted = useMounted();
 
-  const execute = (getPromise) => new Promise((resolve, reject) => {
+  useCallback((getPromise) => new Promise((resolve, reject) => {
     getPromise()
       .then((result) => {
         if (isMounted.current) resolve(result);
       })
       .catch((result) => {
         if (isMounted.current) reject(result);
       });
-  });
-
-  return execute;
+  }), []);
```

Comment:
think you need `getPromise` as dependency in there otherwise you are working with a old `getPromise` function

Reply:
it's within the cached fn (argument)

Reply:
oh yeah. sorry, it's already late 💤